wdtk-datamodel/src/main/java/org/wikidata/wdtk/datamodel/helpers/LexemeDeserializer.java

```diff
@@ -0,0 +1,98 @@
+import com.fasterxml.jackson.databind.type.TypeFactory;
+import org.wikidata.wdtk.datamodel.implementation.*;
+import org.wikidata.wdtk.datamodel.interfaces.FormDocument;
+import org.wikidata.wdtk.datamodel.interfaces.LexemeDocument;
+import org.wikidata.wdtk.datamodel.interfaces.MonolingualTextValue;
+import org.wikidata.wdtk.datamodel.interfaces.SenseDocument;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Custom deserializer, very close to the default serializer generated by Jackson,
+ * only to also accept empty json objects ({}) in place of empty lists for the list of senses.
+ * https://github.com/Wikidata/Wikidata-Toolkit/issues/568
+ */
+public class LexemeDeserializer extends StdDeserializer<LexemeDocumentImpl> {
+
+    public LexemeDeserializer() {
+        super(LexemeDocumentImpl.class);
+    }
+
+    @Override
+    public LexemeDocumentImpl deserialize(JsonParser jsonParser, DeserializationContext deserializationContext) throws IOException, JacksonException {
+        JsonNode node = jsonParser.getCodec().readTree(jsonParser);
+        if (! (node instanceof ObjectNode)) {
+            throw new IOException("Deserializing a lexeme can only be done from a JSON object");
+        }
+        ObjectNode object = (ObjectNode) node;
+
+        String jsonId = object.get("id").asText();
+        String lexicalCategory = object.get("lexicalCategory").asText();
+        String language = object.get("language").asText();
+        TypeFactory typeFactory = deserializationContext.getTypeFactory();
+
+        JsonNode lemmas1 = object.get("lemmas");
+        Map<String, TermImpl> preLemmas = deserializationContext.readTreeAsValue(
+                lemmas1,
+                typeFactory.constructMapType(Map.class, String.class, TermImpl.class));
+        Map<String, MonolingualTextValue> lemmas = null;
+        if (preLemmas != null) {
+            lemmas = preLemmas.entrySet()
+                    .stream()
+                    .collect(Collectors.toMap(Map.Entry::getKey, v -> v.getValue()));
+        }
+
+        Map<String, List<StatementImpl.PreStatement>> claims = deserializationContext.readTreeAsValue(
+                object.get("claims"),
+                typeFactory.constructMapType(Map.class, typeFactory.constructType(String.class),
+                        typeFactory.constructCollectionType(List.class, StatementImpl.PreStatement.class)));
+
+        List<FormDocument> forms = deserializationContext.readTreeAsValue(
+                object.get("forms"),
+                typeFactory.constructCollectionType(List.class, FormDocumentImpl.class));
+
+        List<SenseDocument> senses;
+        if (object.get("senses") instanceof ObjectNode && ((ObjectNode)object.get("senses")).isEmpty()) {
+            // case for https://github.com/Wikidata/Wikidata-Toolkit/issues/568
+            senses = Collections.emptyList();
```

Comment:
Genuine question: couldn't this bug also happen for "forms"?

Reply:
I thought that lexemes always have at least a form, so that list could not be empty, but I am not sure where I dreamed that up… Sigh! At least the custom deserializer will be a bit more worth the boilerplate.